### ISSUE TITLE
Issue when encoding utf characters in comments

### ISSUE THREAD

**Issue by mikefarah** [go-yaml/yaml#578]
I have a sample yaml which I decoded and then re-encoded:

```
value: проверит # проверит
```

the result I get is:

```
value: проверит # ��������
```

String is fine in the nodeValue but gets garbled in the comments :(

**Comment by mikefarah** [comment]
May be related to https://github.com/go-yaml/yaml/issues/538 as the string I had in the comment works fine as a node value...

**Comment by GordonChen13** [comment]
I have the same problem in chinese.
It happens when your language was endoce in utf8 with more than one byte.
It was caused by only add the first byte when parsing multi byte utf8 word.
  
node_test case:
```go
{
"# 一\n# 二\ntrue # 三\n# 四\n# 五\n",
yaml.Node{
	Kind:   yaml.DocumentNode,
		Line:   3,
		Column: 1,
		Content: []*yaml.Node{{
			Kind:        yaml.ScalarNode,
			Value:       "true",
			Tag:         "!!bool",
			Line:        3,
			Column:      1,
			HeadComment: "# 一\n# 二",
			LineComment: "# 三",
			FootComment: "# 四\n# 五",
		}},
	},
}
```
test failed output:
```bash
/usr/local/go/bin/go test -c -o /tmp/___S_TestNodeRoundtrip_in_gopkg_in_yaml_v3 gopkg.in/yaml.v3 #gosetup
/tmp/___S_TestNodeRoundtrip_in_gopkg_in_yaml_v3 -test.v -check.f ^TestNodeRoundtrip$ -check.vv #gosetup
test 0: "# 一\n# 二\ntrue # 三\n# 四\n# 五\n"
  expected comments:
    <DOC> 
      <true> "# 一\n# 二" / "# 三" / "# 四\n# 五"

  obtained comments:
    <DOC> 
      <true> "# \xe4\n# \xe4" / "# \xe4" / "# \xe5\n# \xe4"


c.Assert(node, DeepEquals, item.node)
Expected :yaml.Node = yaml.Node{Kind:0x1, Style:0x0, Tag:"", Value:"", Anchor:"", Alias:(*yaml.Node)(nil), Content:[]*yaml.Node{(*yaml.Node)(0x803480)}, HeadComment:"", LineComment:"", FootComment:"", Line:3, Column:1}
Actual   :yaml.Node = yaml.Node{Kind:0x1, Style:0x0, Tag:"", Value:"", Anchor:"", Alias:(*yaml.Node)(nil), Content:[]*yaml.Node{(*yaml.Node)(0xc0000a2820)}, HeadComment:"", LineComment:"", FootComment:"", Line:3, Column:1}
<Click to see difference>

node_test.go:2279

OOPS: 0 passed, 1 FAILED
--- FAIL: Test (0.00s)
=== RUN   ExampleUnmarshal_embedded
--- PASS: ExampleUnmarshal_embedded (0.00s)
```

**Comment by GordonChen13** [comment]
I make a pull request to v3 branch, and hopefully it will fix your problem too.

**Comment by niemeyer** [comment]
This should be fixed with 913338de. Please let me know if you see similar problems.
